### PR TITLE
Fix run_constrains for clang and llvm packages

### PR DESCRIPTION
**Pull request by isuruf**
Checklist
* [ ] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/gen_patch_json.py [hunk 1]
```diff
@@ -724,16 +733,20 @@ def _extract_track_feature(record, feature_name):
 def main():
     # Step 1. Collect initial repodata for all subdirs.
     repodatas = {}
-    for subdir in tqdm.tqdm(SUBDIRS, desc="Downloading repodata"):
+    if "CF_SUBDIR" in os.environ:
+        subdirs = [os.environ["CF_SUBDIR"]]
```
Comment (beckermr) [inline recipe/gen_patch_json.py:737]: Is this for local debugging? We might want a comment to that effect.

File: recipe/gen_patch_json.py [hunk 2]
```diff
@@ -545,6 +545,15 @@ def _gen_new_index(repodata, subdir):
                     _relax_exact(fn, record, llvm, max_pin="x.x")
                 else:
                     _relax_exact(fn, record, llvm, max_pin="x.x.x")
+
+        if record_name in llvm_pkgs:
```
Comment (beckermr) [inline recipe/gen_patch_json.py:549]: I don't see the def of `llvm_pkgs`

Comment (isuruf) [reply recipe/gen_patch_json.py:549]: It's defined  a few lines above

Comment (beckermr) [reply recipe/gen_patch_json.py:549]: ahhhhh I thought it would be in this diff!

**Comment by isuruf** [comment]
I don't think `.*` was deprecated. Deps without `.*` or a relational operator is deprecated.

**Pull request merged**